src/Material/Menu.elm

```diff
@@ -0,0 +1,523 @@
+> the overall user experience. See the menu component's Material Design
+> specifications page for details.
+
+See also the
+[Material Design Specification]([https://www.google.com/design/spec/components/menus.html).
+
+Refer to
+[this site](https://debois.github.io/elm-mdl/#/menus)
+for a live demo.
+
+# Elm architecture
+@docs Model, model, Action, update, View
+
+# Style
+@docs bottomLeft, bottomRight, topLeft, topRight, unaligned
+-}
+
+import Dict exposing (Dict)
+import Effects exposing (Effects, none)
+import Html.Attributes as Html exposing (..)
+import Html.Events as Html exposing (defaultOptions)
+import Html exposing (..)
+import Json.Decode as Json exposing (Decoder)
+import Json.Encode exposing (string)
+import Material.Helpers exposing (..)
+import String
+import Task
+
+import Material.Menu.Oracle as Oracle exposing (Oracle)
+import Material.Ripple as Ripple
+import Material.Style as Style exposing (Style, cs, cs', css, css', styled)
+
+{-| MDL menu.
+-}
+
+
+-- CONSTANTS
+
+constant :
+  { transitionDurationSeconds  : Float
+  , transitionDurationFraction : Float
+  , closeTimeout               : Float
+  }
+constant =
+  { transitionDurationSeconds  = 0.3
+  , transitionDurationFraction = 0.8
+  , closeTimeout               = 150
+  }
+
+
+-- TODO: Key codes are not implemented yet.
+
+--keycodes =
+--  { enter     = 13
+--  , escape    = 27
+--  , space     = 32
+--  , upArrow   = 38
+--  , downArrow = 40
+--  }
+
+
+-- MODEL
+
+
+{-| Model of the menu; common to all kinds of menus.
+Use `model` to initialise it.
+-}
+
+type alias Model =
+  { alignment : Style
+  , ripple : Bool
+  , items : Dict Int Ripple.Model
+  , open : Maybe Bool --
+  , closing : Bool
+  , oracle : Maybe Oracle
+  }
+
+
+{-| Model initialiser. Call with `True` if the menu items should ripple when clicked, `False` otherwise. Pass one of `bottomLeft`, `bottomRight`, `topLeft`, `topRight` to specify the menu's alignment in relation to the button. Use `unaligned` if you want to align it manually.
+-}
+
+model : Bool -> Style -> Model
+model ripple alignemnt =
+  { alignment = alignemnt
+  , ripple = ripple
+  , items = Dict.empty
+  , open = Nothing
+  , closing = False
+  , oracle = Nothing
+  }
+
+
+{-| Component model styles.
+Used with `model`.
+-}
+
+bottomLeft : Style
+bottomLeft = cs "mdl-menu--bottom-left"
+
+bottomRight : Style
+bottomRight = cs "mdl-menu--bottom-right"
+
+topLeft : Style
+topLeft = cs "mdl-menu--top-left"
+
+topRight : Style
+topRight = cs "mdl-menu--top-right"
+
+unaligned : Style
+unaligned = cs "mdl-menu--unaligned"
+
+
+-- ITEM
+
+
+{-| Item model.
+-}
+
+type alias Item =
+  { html    : Html
+  , enabled : Bool
+  , divider : Bool
+  }
+
+{-| Item constructor. Pass `True` if there should be a divider below this item; `False` otherwise. Pass as second argument `True` if item should be enabled, and pass as third argument any `Html` as content of the menu item.
+-}
+
+item : Bool -> Bool -> Html -> Item
```

Comment:
I like it that an Item contains Html and not just text. 

When you are providing no defaults, I don't think it is necessary to provide a constructor function. (Don't Elm generate a function `Item : Html -> Bool -> Bool -> Item` automatically?)